documentapi/src/main/java/com/yahoo/documentapi/messagebus/protocol/ContentPolicy.java

```diff
@@ -108,13 +108,13 @@ String getRandomTargetSpec(RoutingContext context) {
             while ((targets.total != 0) &&
                    (100 * targets.list.size() / targets.total >= requiredUpPercentageToSendToKnownGoodNodes))
             {
-                int randIndex = randomizer.nextInt(targets.list.size());
-                String targetSpec = getTargetSpec(targets.list.get(randIndex), context);
+                Integer distributor = targets.list.get(randomizer.nextInt(targets.list.size()));
```

Comment:
Is it possible to race with a concurrent removal from `targets.list` so that `randomizer.nextInt(targets.list.size())` returns a value that is greater than the max value accepted by `targets.list.get()`?

Reply:
Yes, it is... Will ponder a fix

Reply:
Just use synchronisation instead? Perhaps with round-robin.